### PR TITLE
[Dash 2294] Add possibility to create a folder

### PR DESCRIPTION
**Pull request by powerczuk**
Form:

screen shot 2016-10-27 at 15 40 36

Error handling:

screen shot 2016-10-27 at 15 41 13

Two new folders `test` and `fewefwfewf`:

screen shot 2016-10-31 at 10 37 14

### CODE REVIEW

File: src/apps/Hosting/Files.js [hunk 1]
```diff
@@ -121,19 +198,65 @@ const HostingFilesView = React.createClass({
...
 
+  renderNewFolderButtons() {
+    const { name, errors, showNewFolderButton } = this.state;
+    const hasErrors = errors.name && errors.name.length;
+    const styles = this.getStyles();
+    const textFieldStyle = hasErrors ? { marginBottom: 22 } : null;
+
+    if (showNewFolderButton && !hasErrors) {
+      return (
+        <div>
```
Comment (Flixow) [inline src/apps/Hosting/Files.js]: It's necessary?

File: src/apps/Hosting/Files.js [hunk 2]
```diff
@@ -121,19 +198,65 @@ const HostingFilesView = React.createClass({
...
+            primary={true}
+            style={styles.newFolderButtons}
+            onTouchTap={this.handleClickNewFolderButton}
+          />
+        </div>
+      );
+    }
+
+    return (
+      <div style={{ display: 'flex', alignItems: 'center' }}>
```
Comment (Flixow) [inline src/apps/Hosting/Files.js]: getStyles()

File: src/apps/Hosting/Files.js [hunk 3]
```diff
@@ -100,6 +152,31 @@ const HostingFilesView = React.createClass({
     return !hasHostingUrl && this.showMissingDomainsSnackbar;
   },
 
+  handleSetNewFolderName(event, name) {
```
Comment (Flixow) [inline src/apps/Hosting/Files.js]: What about `handleNewFolderNameChange`?

**Comment by powerczuk** [comment]
Yes, but problem is resolved by task **[DASH-2329] Fix uploading files to deeper folders**

File: src/apps/Hosting/Files.js [hunk 4]
```diff
@@ -121,19 +202,63 @@ const HostingFilesView = React.createClass({
...
+    return (
+      <div style={styles.newFolderForm}>
+        <TextField
+          fullWidth={true}
+          name="name"
+          value={name}
+          onChange={this.handleNewFolderNameChange}
+          errorText={this.getValidationMessages('name').join(' ')}
+          hintText="Type new folder name"
+          style={{ ...styles.newFolderNameInput, ...textFieldStyle }}
```
Comment (Flixow) [inline src/apps/Hosting/Files.js]: First you create object and next unpack it, make sense?

File: src/apps/Hosting/Files.js [hunk 5]
```diff
@@ -153,35 +278,43 @@ const HostingFilesView = React.createClass({
...
             isLoading={isLoading}
+            isUploading={isUploading}
             items={items}
-            hideDialogs={hideDialogs}
-            errorResponses={errorResponses}
-            handleErrorsButtonClick={HostingFilesActions.finishUploading}
+            lastFileIndex={lastFileIndex}
+            moveDirectoryDown={this.moveDirectoryDown}
+            moveDirectoryUp={this.moveDirectoryUp}
+            previousFolders={previousFolders}
```
Comment (Flixow) [inline src/apps/Hosting/Files.js]: On my mind something went wrong. Is it good idea have that such many props?

File: src/apps/Hosting/Files.js [hunk 6]
```diff
@@ -121,19 +202,63 @@ const HostingFilesView = React.createClass({
...
+          errorText={this.getValidationMessages('name').join(' ')}
+          hintText="Type new folder name"
+          style={{ ...styles.newFolderNameInput, ...textFieldStyle }}
+        />
+        <RaisedButton
+          label="Create"
+          primary={true}
+          style={styles.newFolderButton}
+          onTouchTap={this.handleCreateNewFolder}
+          target="_blank"
```
Comment (Flixow) [inline src/apps/Hosting/Files.js]: Unnecessary

File: src/apps/Hosting/Files.js [hunk 7]
```diff
@@ -121,19 +202,63 @@ const HostingFilesView = React.createClass({
...
+        <TextField
+          fullWidth={true}
+          name="name"
+          value={name}
+          onChange={this.handleNewFolderNameChange}
+          errorText={this.getValidationMessages('name').join(' ')}
+          hintText="Type new folder name"
+          style={{ ...styles.newFolderNameInput, ...textFieldStyle }}
+        />
+        <RaisedButton
```
Comment (Flixow) [inline src/apps/Hosting/Files.js]: You can use one component for both buttons and control it via props. I mean buttons for `Create new folder` and `Create`

File: src/apps/Hosting/Files.js [hunk 8]
```diff
@@ -203,40 +207,31 @@ const HostingFilesView = React.createClass({
...
-          label="Create new folder"
-          primary={true}
-          style={styles.newFolderButton}
-          onTouchTap={this.handleClickNewFolderButton}
-        />
-      );
-    }
+    const createFolderButtonLabel = showNewFolderForm ? 'Create' : 'Create new folder';
+    const createFolderButtonAction = showNewFolderForm ? this.handleCreateFolder : this.handleClickNewFolderButton;
+    const disableNewFolderButton = showNewFolderForm && !name;
```
Comment (Flixow) [inline src/apps/Hosting/Files.js]: What about `isNameFilled` or something like this?

**Pull request merged**